recipes/opentelemetry-cpp/all/conanfile.py

```diff
@@ -195,7 +190,7 @@ def validate(self):
 
     def build_requirements(self):
         if self.options.with_otlp_grpc or self.options.with_otlp_http:
-            self.tool_requires("opentelemetry-proto/1.3.0")
+            self.tool_requires("opentelemetry-proto/1.3.2")
```

Comment:
https://github.com/open-telemetry/opentelemetry-cpp/pull/2991 changes in the documentation seem to imply that there's not need to map 1:1 unless a new version requires a minimum version, so after checking locally, old versions till compile with this, so I've simplified this @ArielGMachado, lmkwyt

Reply:
Without knowing much in depth about the `opentelemetry-cpp`'s internals, what I understand from documentation (https://github.com/open-telemetry/opentelemetry-cpp/blob/main/docs/maintaining-dependencies.md) is that `opentelemetry-cpp` generates code from `opentelemetry-proto` as part of the opentelemetry-cpp build. That are protobuf specs.
Each 3rd party version that uses in `opentelemetry-cpp` is being indicated in the file `third_party_release` of each release.
 When upgrading `opentelemetry-proto` to a newer release code can requires adjustment, so if a version requires code and a specific version of `opentelemetry-proto`, it is indicated in the release. 
According to the [(Release info v1.16.0)](https://github.com/open-telemetry/opentelemetry-cpp/releases/tag/v1.16.0) , was upgraded to `opentelemetry-proto/1.3.1`, and in the [(Release info v1.17.0)](https://github.com/open-telemetry/opentelemetry-cpp/releases/tag/v1.17.0) was upgraded to `opentelemetry-proto/1.3.2`.
I made the change in the recipe according to this principle to ensure the receipe compile each version with the exact dependency version they indicate in the release and not diverge in the Conan recipe from the official version of opentelemetry-cpp project.
I hope this helps you understand the reason for the change.